pkg/cmd/backtest.go

```diff
@@ -597,6 +585,32 @@ var BacktestCmd = &cobra.Command{
 	},
 }
 
+func collectSubscriptionIntervals(environ *bbgo.Environment) (allKLineIntervals map[types.Interval]struct{}, requiredInterval types.Interval, backTestIntervals []types.Interval) {
+	// default extra back-test intervals
+	backTestIntervals = []types.Interval{types.Interval1h, types.Interval1d}
+	// all subscribed intervals
+	allKLineIntervals = make(map[types.Interval]struct{})
+
+	for _, interval := range backTestIntervals {
+		allKLineIntervals[interval] = struct{}{}
+	}
+	// default interval is 1m for all exchanges
+	requiredInterval = types.Interval1m
+	for _, session := range environ.Sessions() {
+		for _, sub := range session.Subscriptions {
+			if sub.Channel == types.KLineChannel {
+				if sub.Options.Interval.Seconds()%60 > 0 {
+					// if any subscription is 1s, then we will use 1s for back-testing
+					requiredInterval = sub.Options.Interval
```

Comment:
use `types.Interval1s` here

Reply:
notice that `Options.Interval = "3s"` is also valid since it's parsable.

Reply:
https://github.com/c9s/bbgo/pull/975#discussion_r986441789